src/software/ai/hl/stp/play/penalty_kick_play_test.cpp

```diff
@@ -12,32 +16,100 @@ class PenaltyKickPlayTest : public SimulatedPlayTestFixture
 {
 };
 
-TEST_F(PenaltyKickPlayTest, test_penalty_kick_play)
+TEST_F(PenaltyKickPlayTest, test_penalty_kick_setup)
 {
-    setBallState(BallState(field().penaltyFriendly(), Vector(0, 0)));
+    setBallState(BallState(field().enemyPenaltyMark(), Vector(0, 0)));
     addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId(
-        {Point(-3, 2.5), Point(-3, 1.5), Point(-3, 0.5), Point(-3, -0.5), Point(-3, -1.5),
-         Point(4.6, -3.1)}));
+        {Point(-2, -2), Point(-3, -1), Point(-3, 0), Point(-3, 1), Point(-3, 2),
+         Point(2, 2.5)}));
     setFriendlyGoalie(0);
     addEnemyRobots(
         TestUtil::createStationaryRobotStatesWithId({field().enemyGoalCenter()}));
     setEnemyGoalie(0);
     setAIPlay(TYPENAME(PenaltyKickPlay));
-    setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::PREPARE_PENALTY_US);
+    setRefereeCommand(RefereeCommand::PREPARE_PENALTY_US, RefereeCommand::NORMAL_START);
 
+    RobotId shooter_id                                               = 5;
     std::vector<ValidationFunction> terminating_validation_functions = {
         // This will keep the test running for 9.5 seconds to give everything enough
         // time to settle into position and be observed with the Visualizer
         // TODO: Implement proper validation
         // https://github.com/UBC-Thunderbots/Software/issues/1396
-        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+        [shooter_id](std::shared_ptr<World> world_ptr,
+                     ValidationCoroutine::push_type& yield) {
+            auto friendly_robots_except_shooter_1_meter_from_ball =
+                [shooter_id](std::shared_ptr<World> world_ptr) {
+                    Point ball_position = world_ptr->ball().position();
+                    for (const auto& robot : world_ptr->friendlyTeam().getAllRobots())
+                    {
+                        if ((robot.id() != shooter_id) &&
+                            ((robot.position() - ball_position).length() < 1))
+                        {
+                            return false;
+                        }
+                    }
+                    return true;
+                };
```

Comment:
See if you can use the `robotAvoidsBall` function here that just went in for the stop play tests

Reply:
this shouldn't be a nonterminating function because this is the setup for the penalty kick (not the actual running of the penalty kick where the 1 meter from ball invariant needs to be held). I can probably add this as a nonterminating test as well for the actual taking of the penalty.

Reply:
used robotsAvoidBall for the nonterminating case in the second test